### PR TITLE
Added a Jenkinsfile for the automated pipeline.

### PR DESCRIPTION
**Pull request by alalazo**
The Jenkinsfile describes the pipeline that will be used in production to run tests daily and collect them into a DB.

### CODE REVIEW

**Comment by alalazo** [comment]
@rmsds @nrichart Can you double check the PR? The pipeline is working fine on this branch. The last commit should make it trigger only for the `master` branch instead of this one. If it looks good I'll merge (and we'll see if it worked tomorrow morning).

**Review by nrichart — APPROVED** [review]
Looks good to me

**Comment by alalazo** [comment]
@nrichart Thanks. I'll merge and keep monitored to see if the cron job is triggered tonight.

File: scripts/run_benchmarks.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# semaphore to indicate that we still need to wait for
+# some job to finish. This file will be deleted as the
+# final step of the process spawned in the background.
+# This is done because otherwise Jenkins (Java) will
+# timeout if the process is not producing any output
+# for 5 mins.
+touch ${benchmarks_dir}/waiting
+
+# Spawn the background process
+echo srun --dependency=afterany:${job_deps} -- sbench collect --db ${db_dir}/benchmarks.db ${benchmarks_dir}
```
Comment (rmsds) [inline scripts/run_benchmarks.sh:61]: Could we avoid the duplication here?
Either put the command into a variable or something like 'set -x; command; set +x`?

File: scripts/run_benchmarks.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+# Update the DB
+db_dir="${HOME}/benchmarks/db"
+raw_results_dir="${HOME}/benchmarks/raw"
+mkdir -p ${db_dir} && mkdir -p ${raw_results_dir}
+
+
+function join_by { local IFS="$1"; shift; echo "$*"; }
+
+job_names="hpl,osu_bw,osu_bibw,osu_latency,osu_alltoall,osu_allreduce"
+job_deps=$(join_by ':' $(squeue --name=${job_names} --format="%.i" | tail -n +2))
```
Comment (rmsds) [inline scripts/run_benchmarks.sh:47]: I think this squeue should probably have a `--user ${USER}` as well.

File: scripts/run_benchmarks.sh (hunk 1 above)
Comment (alalazo) [reply scripts/run_benchmarks.sh:61]: I think both are fair comments. I'll implement them tomorrow in another PR.

**Pull request merged**